decoder.go

```diff
@@ -275,29 +306,39 @@ func (d *Decoder) decode(v reflect.Value, path string, parts []pathPart, values
 				}
 			}
 		} else {
-			// When there's no registered conversion for the custom type, we will check if the type
-			// implements the TextUnmarshaler interface. As the UnmarshalText function should be applied
-			// to the pointer of the type, we convert the value to pointer.
-			if v.CanAddr() {
-				v = v.Addr()
-			}
+			return fmt.Errorf("schema: converter not found for %v", t)
+		}
+	}
+	return nil
+}
 
-			if u, ok := v.Interface().(encoding.TextUnmarshaler); ok {
-				if err := u.UnmarshalText([]byte(val)); err != nil {
-					return ConversionError{
-						Key:   path,
-						Type:  t,
-						Index: -1,
-						Err:   err,
-					}
-				}
+func isTextUnmarshaler(v reflect.Value) (encoding.TextUnmarshaler, bool, bool) {
```

Comment:
I think the return values from this function are becoming confusing now. Maybe it should just return a struct?

```
struct unmarshaller {
    U encoding.TextUnmarhaler
    IsPtr bool
    IsValid bool
}
```